File: LLM-Translations/GPT-4/avatar/Java/Go/codeforces_154_A.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	infile := bufio.NewReader(os.Stdin)
	arr, _ := infile.ReadString('\n')
	arr = strings.TrimSpace(arr)
	N := len(arr)
	set := make(map[string]bool)
	K := 0
	fmt.Scanf("%d\n", &K)
	res := 0

	for k := 0; k < K; k++ {
		input, _ := infile.ReadString('\n')
		input = strings.TrimSpace(input)
		a := 0
		b := 0

		for i := 0; i < N; i++ {
			if arr[i] == input[0] {
				a++
			} else if arr[i] == input[1] {
				b++
			} else {
				res += min(a, b)
				a = 0
				b = 0
			}
		}
		res += min(a, b)
	}
	fmt.Println(res)
}

func min(a, b int) int {
	if a < b {
		return a
	}
	return b
}

// //End of Code

File: LLM-Translations/GPT-4/avatar/Python/C/codeforces_154_A.c
#include <stdio.h>
#include <string.h>

int main() {
    char s[1000], x[2];
    int n, ans = 0, a = 0, b = 0;
    scanf("%s", s);
    scanf("%d", &n);
    for(int i = 0; i < n; i++) {
        scanf("%s", x);
        a = 0, b = 0;
        for(int j = 0; j < strlen(s); j++) {
            if(s[j] == x[0]) {
                a += 1;
            } else if(s[j] == x[1]) {
                b += 1;
            } else {
                ans = ans + ((a < b) ? a : b);
                a = 0, b = 0;
            }
        }
        ans = ans + ((a < b) ? a : b);
    }
    printf("%d", ans);
    return 0;
} // End of Code
